components/SurveyModal.js

```diff
@@ -0,0 +1,566 @@
+        sell your email address.
+      </p>
+      <fieldset>
+        <input
+          style={{ marginBottom: '0.5em' }}
+          type="text"
+          placeholder="First name"
+          name="FNAME"
+          onChange={event => updateForm(event, 'firstName')}
+        />
+        <input
+          style={{ marginBottom: '0.5em' }}
+          type="text"
+          placeholder="Last Name"
+          name="LNAME"
+          onChange={event => updateForm(event, 'lastName')}
+        />
+        <input
+          style={{ marginBottom: '0.5em' }}
+          type="email"
+          placeholder="Email"
+          name="EMAIL"
+          onChange={event => updateForm(event, 'email')}
+        />
+      </fieldset>
+      <div className="tji-modal__actions">
+        <button type="button" onClick={() => skipStep()} className="btn--simple">
+          No Thanks
+        </button>
+        <button
+          type="button"
+          className="btn btn--primary"
+          onClick={() => validateStep(requiredFields, newsletterCallback)}
+        >
+          Continue
+        </button>
+      </div>
+      {stepError !== '' && <p className="tji-modal__form__error">{stepError}</p>}
+    </form>
+  );
+}
+
+Step4.propTypes = {
+  skipStep: PropTypes.func.isRequired,
+  validateStep: PropTypes.func.isRequired,
+  updateForm: PropTypes.func.isRequired,
+  stepError: PropTypes.string,
+};
+
+function Step5(props) {
+  const { cancelForm } = props;
+  return (
+    <form className="tji-modal__form">
+      <h2 className="tji-modal__title">Before you go...</h2>
+      <p className="tji-modal__description">
+        TJI provides its data to all users for free. Please consider making a contribution so we can continue to be a
+        free resource.
+      </p>
+      <div className="tji-modal__actions">
+        <button type="button" onClick={() => cancelForm()} className="btn--simple">
+          Not Now
+        </button>
+        <button type="button" className="btn btn--primary" onClick={() => (window.location.href = 'donate')}>
+          Donate!
+        </button>
+      </div>
+    </form>
+  );
+}
+
+Step5.propTypes = {
+  cancelForm: PropTypes.func.isRequired,
+};
+
+class SurveyModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formActive: null,
+      currentStep: 1,
+      stepError: '',
+      surveyData: {
+        whoami: '',
+        dataSought: '',
+        dataFound: '',
+        firstName: '',
+        lastName: '',
+        email: '',
+        amount: 5,
+      },
+    };
+
+    this.validateStep = this.validateStep.bind(this);
+    this.cancelForm = this.cancelForm.bind(this);
+    this.updateForm = this.updateForm.bind(this);
+    this.skipStep = this.skipStep.bind(this);
+  }
+
+  componentDidMount() {
+    const dataDownloaded = localStorage.getItem('dataDownloaded') === 'true';
```

Comment:
👍👍 to the `localStorage` approach.

Do both `SurveyModal` and `DataDownloadButton` need to know about `dataDownloaded`? It might be nice if just one of them were responsible for deciding if we should render the modal.

Reply:
Agreed, I think just one of these can handle this. I've updated this.

Reply:
Awesome work! 😍